src/xo-app/home/index.js

```diff
@@ -366,13 +373,30 @@ export default class Home extends Component {
 
   _focusFilterInput = () => this.refs.filterInput.focus()
 
+  _saveNewUserFilter = () => {
+    return saveNewUserFilter({
+      type: this.props.type,
+      value: this._getFilter()
+    })
+  }
+
   _renderHeader () {
-    const { filters } = OPTIONS[this.props.type]
+    const { type } = this.props
+    const { filters } = OPTIONS[type]
+
+    const customFilters = (({ preferences } = {}) => {
+      if (!preferences) {
+        return
+      }
+
+      const customFilters = preferences.filters || {}
+      return customFilters[type]
+    })(this.props.user)
```

Comment:
Create a `_getCustomFilters()` which returns the custom helpers for the current type.